### ISSUE TITLE
[JENKINS-18490] Run button not work

### ISSUE THREAD

**Issue by jenkins-infra-bot** [jenkinsci/build-pipeline-plugin#220]
<p>Build Pipeline Plugin throw the next exception when I click in the run button:</p>

<pre>
WARNING: Untrapped Error in Servlet
org.kohsuke.stapler.ForwardToView
	at org.kohsuke.stapler.HttpResponses.forwardToView(HttpResponses.java:213)
	at hudson.model.BuildAuthorizationToken.checkPermission(BuildAuthorizationToken.java:87)
	at hudson.model.AbstractProject.doBuild(AbstractProject.java:1791)
	at au.com.centrumsystems.hudson.plugin.buildpipeline.DownstreamProjectGridBuilder$1.generateResponse(DownstreamProjectGridBuilder.java:172)
	at org.kohsuke.stapler.HttpResponseRenderer$Default.handleHttpResponse(HttpResponseRenderer.java:105)
	at org.kohsuke.stapler.HttpResponseRenderer$Default.generateResponse(HttpResponseRenderer.java:63)
	at org.kohsuke.stapler.Function.renderResponse(Function.java:103)
	at org.kohsuke.stapler.Function.bindAndInvokeAndServeResponse(Function.java:92)
</pre> 

<p>Thanks in advance,</p>

---
<details><summary><i>Originally reported by <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/jrpedrianes.png" title="jrpedrianes's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jrpedrianes">jrpedrianes</a>, imported from: <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-18490" target="_blank">Run button not work</a></i></summary>
<i><ul>
<li><b>status</b>: Resolved
<li><b>priority</b>: Blocker
<li><b>component(s)</b>: build-pipeline-plugin
<li><b>resolution</b>: Fixed
<li><b>resolved</b>: 2013-10-01T15:36:49+00:00
<li><b>votes</b>: 11
<li><b>watchers</b>: 15
<li><b>imported</b>: 20260107-091649
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>Build Pipeline Plugin throw the next exception when I click in the run button:</p>

<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">WARNING: Untrapped Error in Servlet
org.kohsuke.stapler.ForwardToView
	at org.kohsuke.stapler.HttpResponses.forwardToView(HttpResponses.java:213)
	at hudson.model.BuildAuthorizationToken.checkPermission(BuildAuthorizationToken.java:87)
	at hudson.model.AbstractProject.doBuild(AbstractProject.java:1791)
	at au.com.centrumsystems.hudson.plugin.buildpipeline.DownstreamProjectGridBuilder$1.generateResponse(DownstreamProjectGridBuilder.java:172)
	at org.kohsuke.stapler.HttpResponseRenderer$Default.handleHttpResponse(HttpResponseRenderer.java:105)
	at org.kohsuke.stapler.HttpResponseRenderer$Default.generateResponse(HttpResponseRenderer.java:63)
	at org.kohsuke.stapler.Function.renderResponse(Function.java:103)
	at org.kohsuke.stapler.Function.bindAndInvokeAndServeResponse(Function.java:92)
</pre>
</div></div> 

<p>Thanks in advance,</p></pre>
</details>
</details>
<details><summary><i>environment</i></summary>

```
Jenkins: 1.519<br/>
Build Pipeline Plugini: 1.3.4.1
```
</details>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>derjohn:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-18490?focusedId=183068&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-183068">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Keep in mind, that might need to clear the browser cache. I "fixxed" one of my BP probems dropping my browser cache %-/ </p>

<p>The last known working version is 1.509.2 (LTS) plus BP 1.3.3.  All other combinations starting from 1.519 - 1.5.24 or so didn't work. We tried a lot, including nightly builds of the BP plugin. IMHO all BP plugin versions since the switch to jquery don't work.</p>

<p>It might be a good idea to follow <a href="https://github.com/jenkinsci/build-pipeline-plugin/commits/master" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/build-pipeline-plugin/commits/master</a> before starting effort in try &amp; error...</p>



</pre>
</details>

</ul>
</details>

<p>Keep in mind, that might need to clear the browser cache. I "fixxed" one of my BP probems dropping my browser cache %-/ </p>

<p>The last known working version is 1.509.2 (LTS) plus BP 1.3.3.  All other combinations starting from 1.519 - 1.5.24 or so didn't work. We tried a lot, including nightly builds of the BP plugin. IMHO all BP plugin versions since the switch to jquery don't work.</p>

<p>It might be a good idea to follow <a href="https://github.com/jenkinsci/build-pipeline-plugin/commits/master" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/build-pipeline-plugin/commits/master</a> before starting effort in try & error...</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>lgueye:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-18490?focusedId=183609&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-183609">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Same problem here: clicking /view/&lt;pipeline_name&gt;/gridBuilder/build has no effect. It doesn't start a new build.
Also, retrying a failing step in the pipeline doesn't create a new build.</p>
</pre>
</details>

</ul>
</details>

<p>Same problem here: clicking /view/<pipeline_name>/gridBuilder/build has no effect. It doesn't start a new build.<br/>
Also, retrying a failing step in the pipeline doesn't create a new build.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>benmorgan:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-18490?focusedId=183708&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-183708">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Same problem here on 1.527 of Jenkins and 1.3.5 of the pipeline plugin. I see the problem where the initial job is parameterised but it works fine where the initial job is not parameterised.</p>
</pre>
</details>

</ul>
</details>

<p>Same problem here on 1.527 of Jenkins and 1.3.5 of the pipeline plugin. I see the problem where the initial job is parameterised but it works fine where the initial job is not parameterised.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>sajjadmehdi:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-18490?focusedId=185082&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-185082">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>New version of plugin has been released. Build Pipeline 1.4 (Required core 1.521).
Run button works and the retry trigger for failed jobs is taking the parameters. Only feature missing is, the trigger is limited to new/failed jobs. You cannot re-run a successful jobs again.</p>
</pre>
</details>

</ul>
</details>

<p>New version of plugin has been released. Build Pipeline 1.4 (Required core 1.521).<br/>
Run button works and the retry trigger for failed jobs is taking the parameters. Only feature missing is, the trigger is limited to new/failed jobs. You cannot re-run a successful jobs again.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/jglick.png" title="jglick's avatar" /> jglick:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-18490?focusedId=186363&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-186363">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>@suffrage: your problem is unrelated. This button does not support sending parameters to the initial job, so the call to <tt>AbstractProject.doBuild</tt> simply sends a redirect which is swallowed. The <em>Run</em> button works when the initial job is not parameterized. Please file a separate issue rather than reopening or commenting on this one.</p>
</pre>
</details>

</ul>
</details>

<p>@​suffrage: your problem is unrelated. This button does not support sending parameters to the initial job, so the call to <tt>AbstractProject.doBuild</tt> simply sends a redirect which is swallowed. The <em>Run</em> button works when the initial job is not parameterized. Please file a separate issue rather than reopening or commenting on this one.</p>
